lib/prepare.js

```diff
@@ -728,17 +731,21 @@ function updateIconResourceForAdaptive (preparedIcons, resourceMap, platformReso
     // project's config.xml location, so we use it as base path.
     let background;
     let foreground;
+    let monochrome;
     let targetPathBackground;
     let targetPathForeground;
+    let targetPathMonochrome;
 
     for (const density in android_icons) {
         let backgroundVal = '@mipmap/ic_launcher_background';
         let foregroundVal = '@mipmap/ic_launcher_foreground';
+        let monochromeVal = '@mipmap/ic_launcher_monochrome';
 
         background = android_icons[density].background;
         foreground = android_icons[density].foreground;
+        monochrome = android_icons[density].monochrome;
 
-        if (!background || !foreground) {
+        if (!background || !foreground || !monochrome) {
```

Comment:
I think it's still valid to have an adaptive icon without monochrome and this check I think will break that as if `background` and `foreground` are defined, but `monochrome` isn't it will  end up skipping the entire adaptive icon block.

This check was here because you cannot have an adaptive icon without both `background` and `foreground`. If  I recall correctly from the android docs for monochrome icons, it is an addon to adaptive icons, and using a monochrome also requires the adaptive `background` / `foreground` drawables.

So in otherwords, you may have an adaptive icon without monochrome, but if you have monochrome, then you must have the other adaptive icon properties as well.

I think the `|| !monochrome` can be removed from this condition, and I think we will have to have a `if (monochrome)` condition later.

Reply:
With this suggested change however I think for users running on API 33, if they use adaptive icons but with no monochrome icon provided, the icon will end up using the cordova default.

Which might be undesirable...

I guess with what you have now, it will fallback to to the legacy/standard icon (e..g `<icon src="..." />`)?

Maybe before you implement the suggestion, tell me your thoughts.

Reply:
How about I refactor this condition as below:

```
 if (((monochrome) && (!background || !foreground)) || (!background || !foreground)) {
    // This icon isn't an adaptive icon, so skip it
    continue;
}
```

Reply:
```suggestion
        const isAdaptiveIcon = background && foreground;
        const isMonochromeIcon = monochrome && isAdaptiveIcon;
        if (!isMonochromeIcon || !isAdaptiveIcon) {
```
Will something like this work? Might need to test to confirm.

Reply:
I think this is definitely more readable!